src/seq/randseq.jl

```diff
@@ -0,0 +1,88 @@
+# Random Sequence Generator
+# =========================
+#
+# Random sequence generator.
+#
+# This file is a part of BioJulia.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+# abstract sequence generator type
+abstract SequenceGenerator{T}
+
+# Sequence generator of stationary distributions.
+immutable StationaryGenerator{T} <: SequenceGenerator{T}
+    elems::Vector{T}
+    probs::Vector{Float64}
+
+    function StationaryGenerator(elems, probs)
+        @assert sum(probs) ≤ 1
+        @assert length(elems) == length(probs) + 1
+        return new(copy(elems), copy(probs))
+    end
+end
+
+function StationaryGenerator{T,U<:AbstractFloat}(
+        elems::AbstractVector{T}, probs::AbstractVector{U})
+    return StationaryGenerator{T}(elems, probs)
+end
+
+alphatype(::Type{DNANucleotide}) = DNAAlphabet{4}
+alphatype(::Type{RNANucleotide}) = RNAAlphabet{4}
+alphatype(::Type{AminoAcid})     = AminoAcidAlphabet
+
+function randseq{T}(generator::StationaryGenerator{T}, len::Integer)
```

Comment:
Is it worth exporting this function and some of the above types? Would that make it any easier for user code to create a subtype of the `SequenceGenerator` type and use `randseq` with such a type? Or could one use `Bio.Seq.randseq{T}(gen::MyNewStationaryGenerator{T}, ...)` to the same effect?